server/routes/private.js

```diff
@@ -16,7 +19,37 @@ router.use(async function (req, res, next) {
     }
 });
 
-router.get('/test', async(req, res) => {
+router.post('/api/book-appointment', async (req, res) => {
+    // Validate appropriate parameters are passed into the book appointment endpoint.
+    // Verification to make sure appointments are only made for future dates and that the worker is available during that time will be handled by the front end.
+    const paramSchema = Joi.object({
+        studentId: Joi.number().integer().required(),
+        workerTimeslotId: Joi.number().integer().required(),
+        purpose: Joi.string().allow(null).max(300),
```

Comment:
Currently, I set purpose as optional. Should it be mandatory instead? I checked with the design document but it had not specified if purpose is required.

Reply:
Purpose field should be changed to mandatory to help indicate the reason for the student's visit to the service worker in advance. This will help the service worker plan for the meeting accordingly or submit any notes/resources for students to access prior to their visit.

Reply:
Excellent point, thank you! I have updated the code to disallow null for purpose and also updated the sql create table statement accordingly. In addition, tests have been updated to ensure purpose is included and to test various invalid values of purpose.

Reply:
Perfect, thank you for following up on the requested change.